tutorials/en/vectara-beginner-app-tutorial-showcase-the-creation-of-vectara-app-in-legal-use-case.mdx

```diff
@@ -0,0 +1,390 @@
+
+Copy the corpus id and integrate it along with the customer id into your .env file. Additionally, secure the client secret by adding it to the .env file.
+
+---
+
+
+### Importing Necessary Libraries: Setting the Foundation 🧱
+
+Before diving into the intricacies of building our application, it's crucial to import the necessary libraries that will empower our code. This sets the foundation for crafting the interactive web interface, initializing Vectara, streamlining the NLP pipeline, and integrating all these components for a seamless user experience.
+
+```python
+import os
+import tempfile
+import time
+
+import streamlit as st
+from dotenv import load_dotenv
+from langchain.callbacks.streaming_stdout import StreamingStdOutCallbackHandler
+from langchain.chat_models import ChatOpenAI
+from langchain.prompts import PromptTemplate
+from langchain.schema import StrOutputParser
+from langchain.vectorstores import Vectara
+
+load_dotenv()
+
+```
+
+With the libraries imported, we now have the tools at our disposal to embark on the journey of creating an intuitive and powerful application.
+
+---
+
+### **1. Streamlit: Crafting the Web Interface** 🎨
+
+Streamlit is employed to forge a user-friendly web interface for the application, facilitating the easy creation of interactive widgets like text inputs and file uploaders.
+
+```python
+# Sidebar for PDF upload and API keys
+with st.sidebar:
+    st.header("Configuration")
+    uploaded_file = st.file_uploader("Choose a PDF file", type=["pdf"])
+    customer_id = st.text_input("Vectara Customer ID", value=os.getenv("CUSTOMER_ID", ""))
+    api_key = st.text_input("Vectara API Key", value=os.getenv("API_KEY", ""))
+    corpus_id = st.text_input("Vectara Corpus ID", value=str(os.getenv("CORPUS_ID", "")))
+    openai_api_key = st.text_input("OpenAI API Key", value=os.getenv("OPENAI_API_KEY", ""))
+    submit_button = st.button("Submit")
+
+# Constants
+CUSTOMER_ID = customer_id if customer_id else os.getenv("CUSTOMER_ID")
+API_KEY = api_key if api_key else os.getenv("API_KEY")
+CORPUS_ID = int(corpus_id) if corpus_id else int(os.getenv("CORPUS_ID", 0))  # Assuming CORPUS_ID should be an integer
+OPENAI_API_KEY = openai_api_key if openai_api_key else os.getenv("OPENAI_API_KEY")
+    ...
+```
+
+In this snippet, a sidebar is crafted for configuration, where users can upload a PDF file and input necessary API keys.
+
+---
+
+### **2. Vectara: Initialization and Document Retrieval** 📚
+
+Vectara is initialized using a simplified function which conceals the intricacies of API interaction, making it beginner-friendly.
+
+```python
+def initialize_vectara():
+    vectara = Vectara(
+        vectara_customer_id=CUSTOMER_ID,
+        vectara_corpus_id=CORPUS_ID,
+        vectara_api_key=API_KEY
+    )
+    return vectara
+
+vectara_client = initialize_vectara()
+```
+
+In this segment, a `Vectara` client is initialized with essential credentials which will be utilized to interact with Vectara's services.
```

Comment:
This would be the Vectara component, not client, I think. Client is often reserved to mean like a python library for Vectara or some other SDK. No?